sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClientBuilder.java

```diff
@@ -265,53 +302,54 @@ private HttpPipeline getHttpPipeline() {
 
         Configuration userAgentConfiguration = (configuration == null) ? Configuration.NONE : configuration;
 
-        // Closest to API goes first, closest to wire goes last.
-        List<HttpPipelinePolicy> policies = new ArrayList<>();
-
-        policies.add(new BlobDecryptionPolicy(keyWrapper, keyResolver, requiresEncryption));
-        String applicationId = clientOptions.getApplicationId() != null ? clientOptions.getApplicationId()
-            : logOptions.getApplicationId();
-        policies.add(new UserAgentPolicy(applicationId, BLOB_CLIENT_NAME, BLOB_CLIENT_VERSION, userAgentConfiguration));
-        policies.add(new RequestIdPolicy());
-
-        policies.addAll(perCallPolicies);
-        HttpPolicyProviders.addBeforeRetryPolicies(policies);
-        policies.add(BuilderUtils.createRetryPolicy(retryOptions, coreRetryOptions, LOGGER));
-
-        policies.add(new AddDatePolicy());
-
-        // We need to place this policy right before the credential policy since headers may affect the string to sign
-        // of the request.
-        HttpHeaders headers = new HttpHeaders();
-        clientOptions.getHeaders().forEach(header -> headers.put(header.getName(), header.getValue()));
-        if (headers.getSize() > 0) {
-            policies.add(new AddHeadersPolicy(headers));
-        }
-        policies.add(new MetadataValidationPolicy());
+            // Closest to API goes first, closest to wire goes last.
+            List<HttpPipelinePolicy> policies = new ArrayList<>();
 
-        if (storageSharedKeyCredential != null) {
-            policies.add(new StorageSharedKeyCredentialPolicy(storageSharedKeyCredential));
-        } else if (tokenCredential != null) {
-            BuilderHelper.httpsValidation(tokenCredential, "bearer token", endpoint, LOGGER);
-            policies.add(new BearerTokenAuthenticationPolicy(tokenCredential, Constants.STORAGE_SCOPE));
-        } else if (azureSasCredential != null) {
-            policies.add(new AzureSasCredentialPolicy(azureSasCredential, false));
-        } else if (sasToken != null) {
-            policies.add(new AzureSasCredentialPolicy(new AzureSasCredential(sasToken), false));
-        }
+            policies.add(new FetchEncryptionVersionPolicy(getUnencryptedBlobClient(), requiresEncryption));
+            policies.add(new BlobDecryptionPolicy(keyWrapper, keyResolver, requiresEncryption));
```

Comment:
what's the reason to create separate `FetchEncryptionVersionPolicy` ? Could it be absorbed into `BlobDecryptionPolicy`?
Both are public types and there seems to be dependency between them - is this right way of modeling this?

Reply:
I can't recall if there was a technical reason why I separated them. It may have just been easier for me to reason about as I was developing it to keep them separate. I can mess around with merging them and see if I run into any technical issues.

Reply:
Was able to move this logic into BlobDecryptionPolicy